### PR TITLE
Re-organization and Updates for Test Configurations

### PR DESCRIPTION
**Pull request by titusfortner**
Note: this is a PR against a temporary branch that has a Draft PR (#367 )

### Description
Feedback on what the Test Configurations should look like.

I mostly didn't touch the formatting here so I could leave it to Tech Content team to decide what it should look like.

I moved a few things around so everything fits in these main sections. Broken down by how they are referenced in capabilities and what platforms on Sauce they can execute in:
* WebDriver W3C Capabilities (Required)
* Browser W3C Capabilities (Optional)
* Desktop Browser Capabilities: Sauce-Specific (Optional)
* Mobile App Capabilities: Appium Settings (Required)
* Mobile App Capabilities: Sauce-Specific (Optional)
* Desktop and Mobile Capabilities: Sauce-Specific (Optional)
* Virtual Device Capabilities: Sauce-Specific (Optional)

Things Done:
* Added Links to w3c documentation for w3c capabilities
* Added `edgedriverVersion` capability
* Removed the long lists of specifically compatible things (these were outdated)
* Removed `webdriver.remote.quietExceptions` capability reference as the functionality is gone in Selenium 4
* Appended `appium:` to capabilities in Appium settings section
* Moved w3c capabilities out of Appium section since they don't get prepended with `appium:`

TODO:
* Add links to additional sources of configuration documentation (Visual, Appium, Browser specific, etc)
* Add `username` and `accessKey` capabilities

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by nadvolod — APPROVED** [review]
Looks really good @titusfortner

File: docs/test-configuration-options.md [hunk 1]
```diff
@@ -11,80 +11,127 @@ export const Highlight = ({children, color}) => ( <span style={{
...
 
-## Desktop Browser W3C Capabilities (Required)
+## WebDriver W3C Capabilities (Required)
 
-Below are the W3C WebDriver primary test configuration settings for Sauce Labs desktop browser tests.
+Below are the W3C WebDriver primary test configuration settings for Sauce Labs desktop browser tests and mobile tests.
 
-### `browserName`
-__Description__: identifies the user agent.
+### [`browserName`](https://w3c.github.io/webdriver/#dfn-browser-name)
```
Comment (kimsauce) [inline docs/test-configuration-options.md:29]: hi @titusfortner , linking this H3 breaks its `code` font in the sidebar. I tried playing around with it but couldn't get it to work. Do you have any suggestions?

Screen Shot 2021-05-23 at 10 41 52 PM

**Review by kimsauce — COMMENTED** [review]
@titusfortner looks good to me -- thank you for taking the time to make these edits. I'll publish and then send the new version back to you for final sign-off.

File: docs/test-configuration-options.md (hunk 1 above)
Comment (titusfortner) [reply docs/test-configuration-options.md:29]: Yeah, we should probably pull it out and have the link be in the description.

**Comment by titusfortner** [comment]
These changes have been incorporated elsewhere.

**Pull request closed without merging**